graf2d/gpadv7/v7/inc/ROOT/RDrawingAttr.hxx

```diff
@@ -292,6 +293,200 @@ public:
    void SetStyleClasses(const std::vector<std::string> &styles) { fStyleClasses = styles; }
 };
 
+
+
+
+/// Only attributes, which should be inserted into RDrawable, probably should be just part of base class
+
+
+class RDrawableAttributes {
+
+public:
+
+   enum EValuesKind { kBool, kInt, kDouble, kString };
+
+   class Value_t {
+   public:
+      Value_t() = default;
+      virtual ~Value_t() = default;
+      virtual EValuesKind Kind() const = 0;
+      virtual bool Compatible(EValuesKind kind) const { return kind == Kind(); }
+      virtual int GetInt() const { return 0; }
+      virtual double GetDouble() const { return 0; }
+      virtual std::string GetString() const { return ""; }
+   };
+
+   class IntValue_t : public Value_t {
+      int v; ///< integer value
+   public:
+      IntValue_t(int _v = 0) : v(_v) {}
+      EValuesKind Kind() const final { return kInt; }
+      int GetInt() const final { return v; }
+   };
+
+   class DoubleValue_t : public Value_t {
+      double v; ///< double value
+   public:
+      DoubleValue_t(double _v = 0) : v(_v) {}
+      EValuesKind Kind() const final { return kDouble; }
+      double GetDouble() const final { return v; }
+   };
+
+   class StringValue_t : public Value_t {
+      std::string v; ///< string value
+   public:
+      StringValue_t(const std::string _v = "") : v(_v) {}
+      EValuesKind Kind() const final { return kString; }
+      std::string GetString() const final { return v; }
+   };
+
+
+   using Map_t = std::unordered_map<std::string, std::unique_ptr<Value_t>>;
+
+   struct Record_t {
+      std::string type;             ///<! drawable type, not stored in the root file, must be initialized
+      std::string user_class;       ///<  user defined drawable class, can later go inside map
+      Map_t map;                    ///<  JSON_object
```

Comment:
Would `fJsonObject` be a better name?

Reply:
I prefer to use shorter names here, while they will be populated for every instance of RDrawable class.